packages/datadog-instrumentations/src/cassandra-driver.js

```diff
@@ -158,34 +142,33 @@ addHook({ name: 'cassandra-driver', versions: ['3 - 3.2'], file: 'lib/request-ha
     if (!isRequestValid(this, arguments, 3)) {
       return send.apply(this, arguments)
     }
-    const asyncResource = new AsyncResource('bound-anonymous-fn')
-
-    callback = asyncResource.bind(callback)
 
-    arguments[2] = AsyncResource.bind(function () {
-      connectCh.publish({ hostname: handler.connection.address, port: handler.connection.port })
-      return callback.apply(this, arguments)
-    })
+    arguments[2] = function () {
+      startCtx = { hostname: handler.connection.address, port: handler.connection.port, ...startCtx }
+      return connectCh.runStores(startCtx, callback, this, ...arguments)
+    }
 
     return send.apply(this, arguments)
   })
   return RequestHandler
 })
 
-function finish (finishCh, errorCh, error) {
+function finish (finishCh, errorCh, ctx, error) {
   if (error) {
-    errorCh.publish(error)
+    ctx.error = error
+    errorCh.publish(ctx)
   }
-  finishCh.publish()
+  finishCh.runStores(ctx, () => {})
```

Comment:
`runStores` is not really needed here, but I guess it's more convenient right now given how the base class works, so not going to block on this but it's something we might want to improve down the line.